File: Go-Solutions/1431.go
package main

func kidsWithCandies(candies []int, extraCandies int) []bool {
	ret := make([]bool, 0, len(candies))
	max := max(candies)
	for _, v := range candies {
		if v + extraCandies >= max {
			ret = append(ret, true)
		} else {
			ret = append(ret, false)
		}
	}
	return ret
}

func max(candies []int) int {
	if len(candies) == 1 {
		return candies[0]
	}

	max := candies[0]
	for _, v := range candies {
		if v > max {
			max = v
		}
	}
	return max
}

File: Java-Kotlin/src/main/java/array/_1431.java
package array;

import java.util.ArrayList;
import java.util.List;

public class _1431 {

    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        List<Boolean> result = new ArrayList<>();
        int max = getMax(candies);
        for (int num : candies) {
            if (num + extraCandies >= max) {
                result.add(true);
            } else {
                result.add(false);
            }
        }
        return result;
    }

    private int getMax(int[] candies) {
        int max = Integer.MIN_VALUE;
        for (int num : candies) {
            if (num > max) {
                max = num;
            }
        }
        return max;
    }
}
